src/main.rs

```diff
@@ -11,25 +7,36 @@ use std::io::Write;
 use std::process;
 
 use gb_io::reader::SeqReader;
-use regex::Regex;
-
-// Translation table 11 from https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi.
-// Indexing into this array is as follows:
-// T = 0, C = 1, A = 2, G = 3
-// index = (16 * first_base) + (4 * second_base) + third_base
-// So... TTT = 0, TTC = 1, TTA = 2, ... , GGC = 61, GGA = 62, GGG = 63
-const GENETIC_CODE: &str = "FFLLSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG";
-// some codons can be used as Met in the start position
-const STARTS: &str = "---M------**--*----M------------MMMM---------------M------------";
-
-// ERR_BAD_NT is an error value for an invalid nucleotide
+use gb_io::{feature_kind, qualifier_key};
+use once_cell::sync::Lazy;
+
+#[derive(thiserror::Error, Debug)]
+enum Error {
+    #[error("Something went wrong in translation! {0:?}")]
+    BadNucleotide(Vec<usize>),
+
+    #[error("Invalid amino acid: {0}")]
+    InvalidAminoAcid(char),
+}
+
+/// Translation table 11 from https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi.
+/// Indexing into this array is as follows:
+/// T = 0, C = 1, A = 2, G = 3
+/// index = (16 * first_base) + (4 * second_base) + third_base
+/// So... TTT = 0, TTC = 1, TTA = 2, ... , GGC = 61, GGA = 62, GGG = 63
+const GENETIC_CODE: &[u8] = b"FFLLSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG";
+
+/// some codons can be used as Met in the start position
+const STARTS: &[u8] = b"---M------**--*----M------------MMMM---------------M------------";
+
+/// ERR_BAD_NT is an error value for an invalid nucleotide
 const ERR_BAD_NT: usize = 99;
 
-// map a base for indexing the GENETIC_CODE string
-// x (char): base to look up
-// returns: usize
-fn lookup(x: char) -> usize {
-    match x {
+/// map a base for indexing the GENETIC_CODE string
+/// x (char): base to look up
+/// returns: usize
+const fn lookup(x: u8) -> usize {
```

Comment:
The `const` here is not strictly necessary, but it indicates that this function can be executed at compile time and let the compiler perform further optimizations. In this case I don't think it would change much, but if it can be `const` it is better to indicate it because it allows other functions calling it to be `const` too.